android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDataJsonUtil.java

```diff
@@ -153,9 +153,10 @@ public static void processKeyboardPackageUpdateJSON(Context aContext, JSONObject
             for (int i = 0; i < KeyboardController.getInstance().get().size(); i++) {
               Keyboard kbd = KeyboardController.getInstance().getKeyboardInfo(i);
               String version = kbd.getVersion();
+              String updateKMP = kbd.getUpdateKMP();
               if (keyboardID.equalsIgnoreCase(kbd.getKeyboardID()) &&
                   (FileUtils.compareVersions(cloudVersion, version) == FileUtils.VERSION_GREATER) &&
-                  (!kbd.getUpdateKMP().equalsIgnoreCase(cloudKMP))) {
+                  (updateKMP != null && !updateKMP.equalsIgnoreCase(cloudKMP))) {
```

Comment:
Does this fix a known crash?

Suggestion is just cleanup - remove unnecessary parentheses for easier reading of test clauses:

```suggestion
                  updateKMP != null && 
                  !updateKMP.equalsIgnoreCase(cloudKMP)) {
```

Reply:
> Does this fix a known crash?

Pretty [rare one](https://sentry.io/organizations/keyman/issues/3659638286/?project=5983520) 
1 user on 15.0.270 stable and myself